### ISSUE TITLE
AbstractClassCanBeInterface should probably ignore classes with private constructors

### ISSUE THREAD

**Issue by eduardb** [detekt/detekt#9655]
## Expected Behavior of the rule
An abstract class with a private constructor (so not instantiable nor extensible) should (probably?) not be flagged by the `AbstractClassCanBeInterface` rule.

## Context
Such classes have various uses, such as being used as qualifiers for dependency injection frameworks. As such, those cases should not be flagged, or at least there should be an option to configure the rule to ignore them.

Example class: 
```kotlin
abstract class AccountScope private constructor()
```

**Comment by ShubhamWorks02** [comment]
I’d like to work on this.

My plan is to skip `AbstractClassCanBeInterface` reports when an abstract class has only private constructors, since converting it to an interface would change extensibility. I’ll add focused tests for the issue example.

**Comment by cortinico** [comment]
@ShubhamWorks02 assigned to you 👍
